frontend/src/stores/OutputStore.js

```diff
@@ -312,37 +349,257 @@ class OutputStore {
-        let doses = output.dataset.doses;
-        let means = output.dataset.means;
-        let doseRange = _.max(doses) - _.min(doses);
-        let responseRange = _.max(means) - _.min(means);
-        let yArray = [];
-        means.map(item => {
-            let new_item = item * responseRange;
-            yArray.push(new_item);
-        });
-        let xArray = [];
-        cdf.map(item => {
-            let new_item = (item / 100) * doseRange;
-            xArray.push(new_item);
-        });
-        var trace2 = {
-            x: xArray,
-            y: yArray,
+    @observable cdfPlot = [];
+    @action setCDFPlot() {
+        this.cdfPlot = [];
+        var trace1 = {
+            x: this.pValue,
+            y: this.cdf,
+            mode: "markers",
+            type: "scatter",
+            name: "CDF",
+        };
+        this.cdfPlot.push(trace1);
+    }
+    @observable cdfLayout = {
+        showlegend: true,
+        title: {
+            text: "CDF Plot",
+            font: {
+                family: "Courier New, monospace",
+                size: 16,
+            },
+            xref: "paper",
+        },
+        xaxis: {
+            linecolor: "black",
+            linewidth: 1,
+            mirror: true,
+            title: {
+                text: "Percentile",
+                font: {
+                    family: "Courier New, monospace",
+                    size: 14,
+                    color: "#7f7f7f",
+                },
+            },
+        },
+        yaxis: {
+            linecolor: "black",
+            linewidth: 1,
+            mirror: true,
+            title: {
+                text: "CDF",
+                font: {
+                    family: "Courier New, monospace",
+                    size: 14,
+                    color: "#7f7f7f",
+                },
+            },
+        },
+        plot_bgcolor: "",
+        paper_bgcolor: "#eee",
+    };
+
+    @observable currentModel;
+    @observable currentDataset;
+    @observable model_index;
+    @observable param = {};
+    @observable bmdLine = {};
+    @action addBMDLine = model => {
+        this.setResponseModel(model.model_name);
+        let param_variables = this.parameter_variables;
+        let parameters = model.results.parameters;
+        this.param = parameters.reduce(function(result, field, index) {
+            result[param_variables[index]] = field;
+            return result;
+        }, {});
+        let response = this.getBMDLine(model.model_name);
+        this.bmdLine = {
+            x: this.currentOutputObject.dataset.doses,
+            y: response,
             mode: "marker",
             type: "line",
+            name: model.model_name,
         };
-        rootStore.dataStore.addFitCurve(trace2);
     };
-    @action clearPlotData = () => {
-        this.plotData.pop();
+    @action removeBMDLine = () => {
+        this.bmdLine = {};
     };
+
+    @action getBMDLine(model_name) {
```

Comment:
let's refactor this so they're easier to test:

1. create a new standalone `constants.js` and build something like this:

```javascript
const generateLine = {
"Hill":  (doses, params) => {
    return doses.map((dose)=>{
        return this.param.g +(this.param.v * Math.pow(dose, this.param.n)) /
                    (Math.pow(this.param.k, this.param.n) + Math.pow(dose, this.param.n))
})
}
```

2. then, the action is just this:

```javascript
@action getBMDLine(model_name) {
    return constants.generateLine[model_name](this.currentOutputObject.dataset.doses, this.param)
}
```

3. doses shouldn't be just those which a user entered, instead, i think it should be a vector of 100 values  equally spaced between the minmum and maximum dose